### PR TITLE
add v3 transactions and missing fields to receipts & block header

### PR DESCRIPTION
**Pull request by ArielElp**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/131)

### CODE REVIEW

File: api/starknet_api_openrpc.json [hunk 1]
```diff
@@ -1773,6 +1795,107 @@
...
+                                "title": "Max fee",
+                                "$ref": "#/components/schemas/FELT",
+                                "description": "The maximal fee that can be charged for including the transaction"
+                            },
+                            "version": {
+                                "title": "Version",
+                                "description": "Version of the transaction scheme",
+                                "type": "string",
+                                "enum": [
+                                    "0x2"
```
Comment (DelevoXDG) [inline api/starknet_api_openrpc.json:1833]: Shouldn't it be 0x3? 👀

File: api/starknet_api_openrpc.json [hunk 2]
```diff
@@ -2464,9 +3040,10 @@
                         ],
                         "description": "The finality status of the transaction"
                     },
-                    "execution_status": {
-                        "title": "Execution status",
-                        "$ref": "#/components/schemas/TXN_EXECUTION_STATUS"
+                    "execution_reources": {
```
Comment (DelevoXDG) [inline api/starknet_api_openrpc.json:3043]: 👀

**Review by EvyatarO — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/131)* status: 0 of 2 files reviewed, 21 unresolved discussions (waiting on @ArielElp and @DelevoXDG)

___
*[`api/starknet_api_openrpc.json` line 1370 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSxQX3ERHOOkP1UZLc:-NdSxQX3ERHOOkP1UZLd:b-153yq6) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1370)):*
> ```JSON
>                     },
>                     "l1_gas_price": {
>                         "title": "L1 gas price",
> ```

You need to have to gas price fields (eth and strk)
___
*[`api/starknet_api_openrpc.json` line 1799 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSxYnDERyUFk4HppR0:-NdSxYnE77WcfAO189_t:b2nhpgd) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1799)):*
> ```JSON
>             },
>             "DECLARE_TXN_V3": {
>                 "title": "Declare Transaction V2",
> ```

V3?
___
*[`api/starknet_api_openrpc.json` line 1800 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSx_-NA0jdEHLiToHv:-NdSx_-NA0jdEHLiToHw:b2nhpgd) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1800)):*
> ```JSON
>             "DECLARE_TXN_V3": {
>                 "title": "Declare Transaction V2",
>                 "description": "Declare Contract Transaction V2",
> ```

V3?
___
*[`api/starknet_api_openrpc.json` line 1814 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSxaYC8kJi7J05O3OO:-NdSxaYC8kJi7J05O3OP:b-sb2pyw) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1814)):*
> ```JSON
>                             },
>                             "sender_address": {
>                                 "title": "Sender address",
> ```

Don't you want to use the common fields in one place?
___
*[`api/starknet_api_openrpc.json` line 1823 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSxj7X3o9RQOwnmK6e:-NdSxj7Y6rgJJP4LvoE5:b7g324x) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1823)):*
> ```JSON
>                                 "$ref": "#/components/schemas/FELT"
>                             },
>                             "max_fee": {
> ```

no max_fee
___
*[`api/starknet_api_openrpc.json` line 1833 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSxh5r8aYsowcgSynt:-NdSxh5r8aYsowcgSynu:b2ngyx2) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1833)):*
> ```JSON
>                                 "type": "string",
>                                 "enum": [
>                                     "0x2"
> ```

0x3
___
*[`api/starknet_api_openrpc.json` line 1854 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSxntH1dqbtuFuGPXU:-NdSxntH1dqbtuFuGPXV:b-ycyf44) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1854)):*
> ```JSON
>                                 "$ref": "#/components/schemas/RESOURCE_LIMITS"
>                             },
>                             "l2_gas": {
> ```

it's part of v3 but you shouldn't send it on v0.13.0, should we keep it here in the meantime? 
Same for tip, da_mode and init_data
___
*[`api/starknet_api_openrpc.json` line 1880 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSyHhg5RUzxisABlqq:-NdSyHhg5RUzxisABlqr:b-gtrw85) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1880)):*
> ```JSON
>                             }
>                         },
>                         "required": [
> ```

There is no max_fee
l2_gas, tip, paymaster_address,nonce and fee data_availability mode are not requires ATM
___
*[`api/starknet_api_openrpc.json` line 1920 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSyc4BBtAKTqmx5ua5:-NdSyc4BBtAKTqmx5ua6:be0kpwm) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1920)):*
> ```JSON
>                     },
>                     {
>                         "title": "Broadcasted invoke transaction V2",
> ```

Why v2?
___
*[`api/starknet_api_openrpc.json` line 2058 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSyeohDlyEQKc9mPDO:-NdSyeohDlyEQKc9mPDP:b2nhpgd) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L2058)):*
> ```JSON
>                     {
>                         "type": "object",
>                         "title": "Declare txn v2",
> ```

V3?
___
*[`api/starknet_api_openrpc.json` line 2078 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSygib-voPnAB9vqJV:-NdSygib-voPnAB9vqJW:b-oaqmxv) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L2078)):*
> ```JSON
>                             },
>                             "max_fee": {
>                                 "title": "Max fee",
> ```

Not exist in v3
___
*[`api/starknet_api_openrpc.json` line 2106 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSyjlY0KrJvX9RLND1:-NdSyjlY0KrJvX9RLND2:b-kuakfw) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L2106)):*
> ```JSON
>                             },
>                             "l2_gas": {
>                                 "title": "L2 Gas",
> ```

it's part of v3 but you shouldn't send it on v0.13.0, should we keep it here in the meantime? 
(Same for tip, volition fields, init code)
___
*[`api/starknet_api_openrpc.json` line 2131 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSyqkr73K2hOp4KdDr:-NdSyqkr73K2hOp4KdDs:bspson4) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L2131)):*
> ```JSON
>                             }
>                         },
>                         "required": [
> ```

See comment in invoke
___
*[`api/starknet_api_openrpc.json` line 2235 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSyvTU8-rkvdIqz3ij:-NdSyvTVAsoCVawlWA2-:b-dj5u2c) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L2235)):*
> ```JSON
>                     },
>                     "max_fee": {
>                         "title": "Max fee",
> ```

Doesn't exist in v3
___
*[`api/starknet_api_openrpc.json` line 2276 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSz3mMBjzVpg5FWfZN:-NdSz3mMBjzVpg5FWfZO:b-kuakfw) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L2276)):*
> ```JSON
>                     },
>                     "l2_gas": {
>                         "title": "L2 Gas",
> ```

it's part of v3 but you shouldn't send it on v0.13.0, should we keep it here in the meantime? 
(Same for tip, volition fields, init code)
___
*[`api/starknet_api_openrpc.json` line 2301 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSz4k6EF9OJvC6Bows:-NdSz4k6EF9OJvC6Bowt:b-33b6k6) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L2301)):*
> ```JSON
>                     }
>                 },
>                 "required": [
> ```

See comment above
___
*[`api/starknet_api_openrpc.json` line 2493 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSzBnL48vQBlQSyPtX:-NdSzBnL48vQBlQSyPtY:b-ii9mcx) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L2493)):*
> ```JSON
>                                 ]
>                             },
>                             "sender_address": {
> ```

Don't you want to create an object for common fields?
___
*[`api/starknet_api_openrpc.json` line 2523 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSzA1033BrOlA-6mxs:-NdSzA1033BrOlA-6mxt:b-kuakfw) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L2523)):*
> ```JSON
>                                 "$ref": "#/components/schemas/RESOURCE_LIMITS"
>                             },
>                             "l2_gas": {
> ```

it's part of v3 but you shouldn't send it on v0.13.0, should we keep it here in the meantime? 
(Same for tip, volition fields, init code)
___
*[`api/starknet_api_openrpc.json` line 3579 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSzVoV0Cuu4wZonVP-:-NdSzVoV0Cuu4wZonVP0:b-tek9la) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L3579)):*
> ```JSON
>                 "properties": {
>                     "max_amount": {
>                         "title": "max amount",
> ```

max_amount_per_unit?

**Comment by tomek0123456789** [comment]
Just reminding that [`starknet_traceBlockTransactions`](https://github.com/starkware-libs/starknet-specs/blob/v0.5.0/api/starknet_trace_api_openrpc.json#L120) should accept `BLOCK_ID` instead of `BLOCK_HASH`

**Review by ArielElp — COMMENTED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/131)* status: all files reviewed, 21 unresolved discussions (waiting on @DelevoXDG and @EvyatarO)

___
*[`api/starknet_api_openrpc.json` line 1370 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSxQX3ERHOOkP1UZLc:-Ndxjs_e3wbJCQ1RiJBY:b-oomlj9) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1370)):*
<details><summary><i>Previously, EvyatarO (Evyatar) wrote…</i></summary><blockquote>

You need to have to gas price fields (eth and strk)
</blockquote></details>

Added (still need to add the exact unit in the description; are we going with 10^-18?)
___
*[`api/starknet_api_openrpc.json` line 1799 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSxYnDERyUFk4HppR0:-NdxkFq6Ajp-dEe4ur4G:b-896fix) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1799)):*
<details><summary><i>Previously, EvyatarO (Evyatar) wrote…</i></summary><blockquote>

V3?
</blockquote></details>

Done.
___
*[`api/starknet_api_openrpc.json` line 1800 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSx_-NA0jdEHLiToHv:-NdxkGVK0dajTm8BLoIZ:b-896fix) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1800)):*
<details><summary><i>Previously, EvyatarO (Evyatar) wrote…</i></summary><blockquote>

V3?
</blockquote></details>

Done.
___
*[`api/starknet_api_openrpc.json` line 1823 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSxj7X3o9RQOwnmK6e:-NdxkHux2Ug4N3tQd0EQ:b-896fix) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1823)):*
<details><summary><i>Previously, EvyatarO (Evyatar) wrote…</i></summary><blockquote>

no max_fee
</blockquote></details>

Done.
___
*[`api/starknet_api_openrpc.json` line 1833 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#gh-1313868894:-NdxkPxGEY-gjisLX2Kt:b-896fix) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1833)):*
<details><summary><i>Previously, DelevoXDG (Maksim Zdobnikau) wrote…</i></summary><blockquote>

Shouldn't it be 0x3? 👀
</blockquote></details>

Done.
___
*[`api/starknet_api_openrpc.json` line 1833 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSxh5r8aYsowcgSynt:-NdxkQP0-bDfLPzthQYk:b-896fix) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1833)):*
<details><summary><i>Previously, EvyatarO (Evyatar) wrote…</i></summary><blockquote>

0x3
</blockquote></details>

Done.
___
*[`api/starknet_api_openrpc.json` line 1854 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSxntH1dqbtuFuGPXU:-Ndxfi5d0RPqII-OEwUA:bwor7um) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1854)):*
<details><summary><i>Previously, EvyatarO (Evyatar) wrote…</i></summary><blockquote>

it's part of v3 but you shouldn't send it on v0.13.0, should we keep it here in the meantime? 
Same for tip, da_mode and init_data
</blockquote></details>

Removed all. We should add them as required in the corresponding version.
___
*[`api/starknet_api_openrpc.json` line 1880 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSyHhg5RUzxisABlqq:-NdxfwH75Bn4uA_3dE6O:b-896fix) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1880)):*
<details><summary><i>Previously, EvyatarO (Evyatar) wrote…</i></summary><blockquote>

There is no max_fee
l2_gas, tip, paymaster_address,nonce and fee data_availability mode are not requires ATM
</blockquote></details>

Done.
___
*[`api/starknet_api_openrpc.json` line 1920 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSyc4BBtAKTqmx5ua5:-Ndxg6wx-YT15OByP9a5:b-896fix) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1920)):*
<details><summary><i>Previously, EvyatarO (Evyatar) wrote…</i></summary><blockquote>

Why v2?
</blockquote></details>

Done.
___
*[`api/starknet_api_openrpc.json` line 2058 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSyeohDlyEQKc9mPDO:-NdxgQjTAxQw-h-jwYYh:b-896fix) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L2058)):*
<details><summary><i>Previously, EvyatarO (Evyatar) wrote…</i></summary><blockquote>

V3?
</blockquote></details>

Done.
___
*[`api/starknet_api_openrpc.json` line 2078 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSygib-voPnAB9vqJV:-NdxgRu-1vpfXOfYgMEO:b-896fix) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L2078)):*
<details><summary><i>Previously, EvyatarO (Evyatar) wrote…</i></summary><blockquote>

Not exist in v3
</blockquote></details>

Done.
___
*[`api/starknet_api_openrpc.json` line 2106 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSyjlY0KrJvX9RLND1:-Ndxgj-M1MlDp_VIR0Al:b-4i76gp) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L2106)):*
<details><summary><i>Previously, EvyatarO (Evyatar) wrote…</i></summary><blockquote>

it's part of v3 but you shouldn't send it on v0.13.0, should we keep it here in the meantime? 
(Same for tip, volition fields, init code)
</blockquote></details>

Deleted until it's actually relevant
___
*[`api/starknet_api_openrpc.json` line 2131 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSyqkr73K2hOp4KdDr:-Ndxgrl23ZcizKnCILyh:b-896fix) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L2131)):*
<details><summary><i>Previously, EvyatarO (Evyatar) wrote…</i></summary><blockquote>

See comment in invoke
</blockquote></details>

Done.
___
*[`api/starknet_api_openrpc.json` line 2235 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSyvTU8-rkvdIqz3ij:-NdxgsVf7mfbXHjcR557:b-896fix) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L2235)):*
<details><summary><i>Previously, EvyatarO (Evyatar) wrote…</i></summary><blockquote>

Doesn't exist in v3
</blockquote></details>

Done.
___
*[`api/starknet_api_openrpc.json` line 2276 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSz3mMBjzVpg5FWfZN:-NdxgtXn2YrVgpbIXW6i:b-896fix) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L2276)):*
<details><summary><i>Previously, EvyatarO (Evyatar) wrote…</i></summary><blockquote>

it's part of v3 but you shouldn't send it on v0.13.0, should we keep it here in the meantime? 
(Same for tip, volition fields, init code)
</blockquote></details>

Done.
___
*[`api/starknet_api_openrpc.json` line 2301 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSz4k6EF9OJvC6Bows:-Ndxgu5kD2dDardJ4drK:b-896fix) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L2301)):*
<details><summary><i>Previously, EvyatarO (Evyatar) wrote…</i></summary><blockquote>

See comment above
</blockquote></details>

Done.
___
*[`api/starknet_api_openrpc.json` line 2523 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSzA1033BrOlA-6mxs:-Ndxh-FxAae_tU6K3AtG:b-896fix) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L2523)):*
<details><summary><i>Previously, EvyatarO (Evyatar) wrote…</i></summary><blockquote>

it's part of v3 but you shouldn't send it on v0.13.0, should we keep it here in the meantime? 
(Same for tip, volition fields, init code)
</blockquote></details>

Done.
___
*[`api/starknet_api_openrpc.json` line 3579 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSzVoV0Cuu4wZonVP-:-NdxhBUN1901-x_9oNBT:b-ojy49d) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L3579)):*
<details><summary><i>Previously, EvyatarO (Evyatar) wrote…</i></summary><blockquote>

max_amount_per_unit?
</blockquote></details>

What's max_amount_per_unit? ATM there's max_amount and max_price_per_unit

**Review by EvyatarO — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/131)* status: all files reviewed, 6 unresolved discussions (waiting on @ArielElp and @DelevoXDG)

___
*[`api/starknet_api_openrpc.json` line 1370 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSxQX3ERHOOkP1UZLc:-Ndxx8yRCClxc79E_jCL:bsshpza) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1370)):*
<details><summary><i>Previously, ArielElp wrote…</i></summary><blockquote>

Added (still need to add the exact unit in the description; are we going with 10^-18?)
</blockquote></details>

We are going w/ 10^-18, still no name yet.
___
*[`api/starknet_api_openrpc.json` line 3579 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSzVoV0Cuu4wZonVP-:-Ndxyld15zizRxMGKZ_k:b-hjmjna) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L3579)):*
<details><summary><i>Previously, ArielElp wrote…</i></summary><blockquote>

What's max_amount_per_unit? ATM there's max_amount and max_price_per_unit
</blockquote></details>

You are correct.

**Review by ArielElp — COMMENTED** [review]
> Dismissed @EvyatarO from 4 discussions.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/131)* status: 0 of 3 files reviewed, 3 unresolved discussions (waiting on @DelevoXDG and @EvyatarO)

___
*[`api/starknet_api_openrpc.json` line 1814 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSxaYC8kJi7J05O3OO:-Ndy2pFDEjdBNC05kPd_:b-fbum8w) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L1814)):*
<details><summary><i>Previously, EvyatarO (Evyatar) wrote…</i></summary><blockquote>

Don't you want to use the common fields in one place?
</blockquote></details>

These objects used to be less readable when non-flat, I rather keep it flat for now until we can completely remove old transaction types.
___
*[`api/starknet_api_openrpc.json` line 2493 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-NdSzBnL48vQBlQSyPtX:-Ndy36wp3G9EG6hS_36P:b-fbum8w) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L2493)):*
<details><summary><i>Previously, EvyatarO (Evyatar) wrote…</i></summary><blockquote>

Don't you want to create an object for common fields?
</blockquote></details>

These objects used to be less readable when non-flat, I rather keep it flat for now until we can completely remove old transaction types.
___
*[`api/starknet_api_openrpc.json` line 3043 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#gh-1313870572:-Ndy3Cj31awIF05HFJpu:b3kre) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L3043)):*
<details><summary><i>Previously, DelevoXDG (Maksim Zdobnikau) wrote…</i></summary><blockquote>

👀
</blockquote></details>

?

File: api/starknet_api_openrpc.json (hunk 2 above)
Comment (DelevoXDG) [reply api/starknet_api_openrpc.json:2924]: Typo, I think it should be "execution_resources"

**Review by ArielElp — COMMENTED** [review]
:lgtm:

> Reviewed 2 of 3 files at r3, 1 of 1 files at r4.
> Dismissed @DelevoXDG from a discussion.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#-:-NeESsNb6LjUkp3NW8VL:bnfp4nl)* status: 2 of 3 files reviewed, all discussions resolved (waiting on @ArielElp)

___
*[`api/starknet_api_openrpc.json` line 3043 at r1](https://reviewable.io/reviews/starkware-libs/starknet-specs/131#gh-1313870572:-NeERCwZ1G1103Tv1x-P:b-s4envi) ([raw file](https://github.com/starkware-libs/starknet-specs/blob/db4688128328cf87204199178da98abf237bc5d1/api/starknet_api_openrpc.json#L3043)):*
<details><summary><i>Previously, DelevoXDG (Maksim Zdobnikau) wrote…</i></summary><blockquote>

Typo, I think it should be "execution_resources"
</blockquote></details>

Ah true, thanks!

**Pull request merged**